broken C generated: member reference base type is not a structure or union

I get an error compiling this code:

### Example
```
import asynchttpserver, asyncdispatch
import strformat

proc main() =
    let local = "123"

    proc serveIndex(req: Request) {.async, gcsafe.} =
        await req.respond(Http200, &"{local}")

    proc serve404(req: Request) {.async, gcsafe.} =
        echo req.url.path
        await req.respond(Http404, "not found")

    proc serve(req: Request) {.async, gcsafe.} =
        let handler = case req.url.path:
            of "/":
                serveIndex
            else:
                serve404
        await handler(req)
    
    let server = newAsyncHttpServer()
    waitFor server.serve(Port(8080), serve, address = "127.0.0.1")

when isMainModule:
    main()
```

### Current Output
```
Error: execution of an external compiler program 'clang -c  -w  -I/Users/willi/.choosenim/toolchains/nim-0.19.0/lib -o /Users/willi/.cache/nim/repro_d/destiny_builder_repro.c.o /Users/willi/.cache/nim/repro_d/destiny_builder_repro.c' failed with exit code: 1

/Users/willi/.cache/nim/repro_d/destiny_builder_repro.c:1311:87: error: member reference base type 'tyObject_FuturecolonObjectType__te3W2Tqi7xuJ7rlPtg9al5w *(tyObject_Request_1Wfxxtqzahm0R6lueM9aQmQ *)' (aka 'struct tyObject_FuturecolonObjectType__te3W2Tqi7xuJ7rlPtg9al5w *(struct tyObject_Request_1Wfxxtqzahm0R6lueM9aQmQ *)') is not a structure or union
                                asgnRef((void**) (&(*colonenvP_).handler1.ClE_0), serve404_cMCA9cDbggCQHYvdZAVf2hQ.ClE_0);
                                                                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/Users/willi/.cache/nim/repro_d/destiny_builder_repro.c:1312:68: error: member reference base type 'tyObject_FuturecolonObjectType__te3W2Tqi7xuJ7rlPtg9al5w *(tyObject_Request_1Wfxxtqzahm0R6lueM9aQmQ *)' (aka 'struct tyObject_FuturecolonObjectType__te3W2Tqi7xuJ7rlPtg9al5w *(struct tyObject_Request_1Wfxxtqzahm0R6lueM9aQmQ *)') is not a structure or union
                                (*colonenvP_).handler1.ClP_0 = serve404_cMCA9cDbggCQHYvdZAVf2hQ.ClP_0;
                                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
2 errors generated.
```

### Additional Information

This is on macOS X Mojave on stable 0.19.

--- a/tests/ccgbugs/t9655.nim
+++ b/tests/ccgbugs/t9655.nim
@@ -1,0 +1,30 @@
+discard """
+  action: "compile"
+"""
+
+import std/[asynchttpserver, asyncdispatch]
+import std/[strformat]
+
+proc main() =
+  let local = "123"
+
+  proc serveIndex(req: Request) {.async, gcsafe.} =
+    await req.respond(Http200, &"{local}")
+
+  proc serve404(req: Request) {.async, gcsafe.} =
+    echo req.url.path
+    await req.respond(Http404, "not found")
+
+  proc serve(req: Request) {.async, gcsafe.} =
+    let handler = case req.url.path:
+      of "/":
+        serveIndex
+      else:
+        serve404
+    await handler(req)
+
+  let server = newAsyncHttpServer()
+  waitFor server.serve(Port(8080), serve, address = "127.0.0.1")
+
+when isMainModule:
+  main()

